corehq/apps/app_manager/suite_xml.py

```diff
@@ -1950,82 +1950,132 @@ def get_target_module(case_type, module_id, with_product_details=False):
+                    changed_id = {
+                        case_tag: {
+                            'old_id': datum['datum'].id,
+                            'new_id': new_id,
+                        }
+                    }
                     datum['datum'].id = new_id
+            return changed_id
 
-                changed_ids[case_tag] = {
-                    "old_id": this_datum_.id,
-                    "new_id": parent_datum_.id
-                }
-            this_datum_.id = parent_datum_.id
+        def get_changed_id(this_datum_meta_, parent_datum_meta_):
+            """
+            Maps IDs in the child module to IDs in the parent module
 
-        root_module = module.root_module
-        root_datums = []
-        if root_module and root_module.module_type == 'basic':
-            # For advanced modules the onus is on the user to make things work by loading the correct cases and
-            # using the correct case tags.
-            try:
-                # assume that all forms in the root module have the same case management
-                root_module_form = root_module.get_form(0)
-            except FormNotFoundException:
-                pass
-            else:
-                if root_module_form.requires_case():
-                    root_datums.extend(self.get_datum_meta_module(root_module))
-                root_datums.extend(self.get_new_case_id_datums_meta(root_module_form))
+            e.g. The case with the ID "parent_id" in the child module has the
+            ID "case_id" in the parent module.
+            """
+            changed_id = {}
+            action = this_datum_meta_['action']
+            if action:
+                case_tag = getattr(action, 'case_tag', '')
+                changed_id = {
+                    case_tag: {
+                        "old_id": this_datum_meta_['datum'].id,
+                        "new_id": parent_datum_meta_['datum'].id
+                    }
+                }
+            return changed_id
 
-        if root_datums:
+        def get_datums(module_):
+            """
+            Return the datums of the first form in the given module
+            """
+            datums_ = []
+            if module_ and module_.module_type == 'basic':
+                # For advanced modules the onus is on the user to make things work by loading the correct cases and
+                # using the correct case tags.
+                try:
+                    # assume that all forms in the module have the same case management
+                    form = module_.get_form(0)
+                except FormNotFoundException:
+                    pass
+                else:
+                    if form.requires_case():
+                        datums_.extend(self.get_datum_meta_module(module_))
+                    datums_.extend(self.get_new_case_id_datums_meta(form))
+                    datums_.extend(self.get_extra_case_id_datums(form))
```

Comment:
Adding this line (commit f3aa4da) will hopefully simplify an awful merge conflict.